scripts/ui_release.sh

```diff
@@ -74,13 +77,81 @@ function bumpVersion() {
   fi
   # upgrade the @perses-dev/* dependencies on all packages
   for workspace in ${workspaces}; do
-    sed -E -i "" "s|(\"@perses-dev/.+\": )\".+\"|\1\"\^${version}\"|" "${workspace}"/package.json
+    # sed -i syntax is different on mac and linux
+    if [[ "$OSTYPE" == "darwin"* ]]; then
+      sed -E -i "" "s|(\"@perses-dev/.+\": )\".+\"|\1\"\^${version}\"|" "${workspace}"/package.json
+    else
+      sed -E -i "s|(\"@perses-dev/.+\": )\".+\"|\1\"\^${version}\"|" "${workspace}"/package.json
+    fi
   done
 
   # increase the version on all packages
   npm version "${version}" --workspaces
 }
 
+# Validates branch name and extracts snapshot name from it.
+function getBranchSnapshotName() {
+  branch="${1}"
+
+  snapshotRegex="^snapshot\/[-a-zA-Z0-9]+$"
```

Comment:
Forcing kebab-case here for consistency and because that's a safe syntax for npm tags/versions. Let me know if anyone has strong feelings about supporting something more complex.